includes/rest-api/class-sensei-rest-api-extensions-controller.php

```diff
@@ -311,14 +367,23 @@ private function create_extensions_response( array $plugins, string $response_ke
 			function ( $plugin ) use ( $wccom_connected ) {
 				$plugin->price      = html_entity_decode( $plugin->price );
 				$plugin->image      = $plugin->image_large ?? 'https://senseilms.com/wp-content/uploads/2021/04/' . $plugin->product_slug . '.png';
-				$plugin->can_update = ! $plugin->wccom_product_id || ( $wccom_connected && ! $plugin->wccom_expired );
+				$plugin->can_update = empty( $plugin->wccom_product_id ) || ( $wccom_connected && ! $plugin->wccom_expired );
 				return $plugin;
 			},
 			$plugins
 		);
 
-		$response_json                  = [ 'wccom_connected' => $wccom_connected ];
-		$response_json[ $response_key ] = array_values( $mapped_plugins );
+		$response_json = [];
+
+		if ( $main_fetch ) {
```

Comment:
What would be the issue if we included everything in all requests? As WC.com data are cached I don't see this creating a performance issue. If it is needed maybe we should consider renaming `main_fetch` to `full_response`?

Reply:
It's just to remove not unnecessary data from those fetches and make it lighter (mainly about the layout that is not very small). Variable renamed here: https://github.com/Automattic/sensei/pull/4214/commits/05824abc3fabe24c8cf6564c9e1c178ad64c381a

Reply:
In general I think that we should avoid optimizations which add more complexity to the code if there are no performance issues. When reading this part of the code, I got the impression that the frontend behaved differently when the extra information is omitted. 

On the other hand the extra complexity is not much so we can leave it as it is.